src/db/db.c

```diff
@@ -568,29 +568,37 @@ dbReplayWait(Db *this, const String *targetLsn, TimeMSec timeout)
         dbExec(this, STRDEF("checkpoint"));
 
         // On PostgreSQL >= 9.6 the checkpoint location can be verified
-        //
-        // ??? We have seen one instance where this check failed.  Is there any chance that the replayed position could be ahead of
-        // the checkpoint recorded in pg_control?  It seems possible since it would be safer if the checkpoint in pg_control was
-        // behind rather than ahead, so add a loop to keep checking until the checkpoint has been recorded or timeout.
+        // Loop until lsn has been reached or timeout
+        wait = waitNew(timeout);
+        targetReached = false;
+        const String *checkpointLsn = NULL;
+
         if (dbPgVersion(this) >= PG_VERSION_96)
         {
-            // Build the query
-            const String *query = strNewFmt(
-                "select (checkpoint_%s > '%s')::bool as targetReached,\n"
-                "       checkpoint_%s::text as checkpointLsn\n"
-                "  from pg_catalog.pg_control_checkpoint()",
-                lsnName, strZ(targetLsn), lsnName);
-
-            // Execute query
-            VariantList *row = dbQueryRow(this, query);
+            do
+            {
+                // Build the query
+                const String *query = strNewFmt(
+                    "select checkpoint_%s::text,\n"
+                    "       (checkpoint_%s > '%s')::bool as targetReached\n"
+                    "  from pg_catalog.pg_control_checkpoint() as checkpointLsn",
+                    lsnName, lsnName, strZ(targetLsn));
```

Comment:
The new query is the same as the old query, just the select fields have been swapped (and `as  checkpointLsn` is now in the wrong place). This also causes churn in the macro. Can we restore the original query and reduce the churn?

Reply:
I preferred to adjust the query so it is consistent with the replay query. 
I guess I can go back to previous query if you'd prefer...

Reply:
Should be solved with 6b4b253.

Reply:
If we wanted to make the queries look more like each other (which I'm not sure is needed) we'd want to do that in a separate patch. Here it just looks like noise.

Update looks good.

Reply:
> If we wanted to make the queries look more like each other (which I'm not sure is needed) we'd want to do that in a separate patch. Here it just looks like noise.

Right, understood :-)

> Update looks good.

Cool, thanks :-)